### PR TITLE
[operator] Introduce `dns.domains` field

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area scalability
/area usability
/kind api-change

**What this PR does / why we need it**:
This PR brings several improvements to the Gardener Operator:
- Add a new `dns.domains` field to replace `dns.domain`. This allows to expose the Virtual Kube-Apiserver via multiple domains.
- All domains configured in `dns.domains` are considered for Istio's SNI configuration as well as for the server certificate of Kube-Apiserver
- Considers the additional `gardener.` subdomain for the SNI configuration (eadcf607ad083134bbe371205cbad69c2198295a)
- Fixes a small bug in the validation code along the way (40d120ff99e0d5bff33794b40ec6a82eb8213535)

**Which issue(s) this PR fixes**:
Part of #7016 

**Special notes for your reviewer**:

**Release note**:

```breaking operator
A new field `.spec.virtualCluster.dns.domains` was added to the `Garden` API. This field allows to expose the `kube-apiserver` of the virtual cluster via multiple domains. Earlier, the API only accepted one domain name via `.spec.virtualCluster.dns.domain`.
⚠️ With this change `.spec.virtualCluster.dns.domain` is deprecated and will be removed in the next release. Please update your `Garden` resource to the new `.spec.virtualCluster.dns.domains` field by removing the existing domain configuration from `dns.domain` and add it as the first entry of `dns.domains`.
```

### CODE REVIEW

File: pkg/apis/operator/v1alpha1/validation/validation.go [hunk 1]
```diff
@@ -78,6 +78,22 @@ func ValidateGardenUpdate(oldGarden, newGarden *operatorv1alpha1.Garden) field.E
 func validateVirtualClusterUpdate(oldVirtualCluster, newVirtualCluster operatorv1alpha1.VirtualCluster, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
+	// Check if old Domain field was properly transferred to Domains field.
+	if oldVirtualCluster.DNS.Domain != nil && len(newVirtualCluster.DNS.Domains) > 0 && *oldVirtualCluster.DNS.Domain != newVirtualCluster.DNS.Domains[0] {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("dns", "domains").Index(0), newVirtualCluster.DNS.Domains[0], "first entry must be previously used domain from .spec.virtualCluster.dns.domain"))
```
Comment (rfranzke) [inline pkg/apis/operator/v1alpha1/validation/validation.go]:
```suggestion
		allErrs = append(allErrs, field.Invalid(fldPath.Child("dns", "domains").Index(0), newVirtualCluster.DNS.Domains[0], "first entry must be the same as previously used in .spec.virtualCluster.dns.domain"))
```

**Review by rfranzke — COMMENTED** [review]
Nice PR!

File: pkg/operator/controller/garden/components.go [hunk 2]
```diff
@@ -154,7 +154,14 @@ func (r *Reconciler) instantiateComponents(
 	if err != nil {
 		return
 	}
-	c.virtualGardenGardenerAccess = r.newGardenerAccess(secretsManager, garden.Spec.VirtualCluster.DNS.Domain)
+
+	var accessDomain string
+	if domains := garden.Spec.VirtualCluster.DNS.Domains; len(domains) > 0 {
+		accessDomain = domains[0]
+	} else {
+		accessDomain = *garden.Spec.VirtualCluster.DNS.Domain
```
Comment (acumino) [inline pkg/operator/controller/garden/components.go:162]: Nil check.

File: test/integration/operator/garden/garden_test.go [hunk 3]
```diff
@@ -204,7 +204,7 @@ var _ = Describe("Garden controller tests", func() {
 				},
 				VirtualCluster: operatorv1alpha1.VirtualCluster{
 					DNS: operatorv1alpha1.DNS{
-						Domain: "virtual-garden.local.gardener.cloud",
+						Domain: pointer.String("virtual-garden.local.gardener.cloud"),
```
Comment (acumino) [inline test/integration/operator/garden/garden_test.go]: Same as other tests?
```suggestion
						Domains: []string{"virtual-garden.local.gardener.cloud"},
```

File: pkg/operator/controller/garden/components.go (hunk 2 above)
Comment (timuthy) [reply pkg/operator/controller/garden/components.go:162]: Shouldn't be required here. It's guaranteed by validation that at least one domain is configured.

**Review by acumino — COMMENTED** [review]
/retest
/lgtm

**Pull request merged**